### DISCUSSION TITLE
[@types/react] version `17.0.56` introduced breaking change

### DISCUSSION THREAD

**Discussion by Hotell** [DefinitelyTyped/DefinitelyTyped#64999] [Issues with a @types package] [answered]
See original issue to learn more https://github.com/microsoft/fluentui/issues/27425

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
Could you trim this down to just using `@types/react` so that we understand what you considered "breaking" in the recent version?

**Comment by eps1lon** [comment, marked as answer]
It sounds like you have an interface that extended `HTMLAttributes` but also declared RDFa props. That declaration is not compatible with the RDFa prop.

The solution is to omit that prop e.g.

```diff
-interface TooltipProps extendsHTMLAttributes {
+interface TooltipProps extends Omit<HTMLAttributes, 'content'> {
   content?: Element | Element[] | string | undefined
 }
```

These are the sort of breaking changes we and any other type declaration has to make occasionally. Otherwise every change would need to be a SemVer major because type changes are generally backwards incompatible.

**Reply by Hotell** [reply]
right I also  thought about that workaround. but still it's a breaking change. 

I understand how brittle DT type changes can be, but it would be great to do some sort of announcement if you're aware of those in advance.

thx

**Reply by eps1lon** [reply]
> I understand how brittle DT type changes can be

Any types change. This isn't specific to DT.

> but it would be great to do some sort of announcement if you're aware of those in advance.

Changes are released immediately. It's hard to imagine what might break and how to migrate. For this one I could've probably thought about it and added the proper fix to the PR description. But generally, we can't cover every usage.

It would be awesome to get a changelog in the DT types. So if you want to champion that change, I'm happy to assist.
